src/Filter/ODM/AbstractFilter.php

```diff
@@ -8,74 +8,39 @@
 
 namespace Laminas\ApiTools\Doctrine\QueryBuilder\Filter\ODM;
 
-use DateTime;
 use Laminas\ApiTools\Doctrine\QueryBuilder\Filter\FilterInterface;
+use Laminas\ApiTools\Doctrine\QueryBuilder\Filter\TypeCastInterface;
 
 abstract class AbstractFilter implements FilterInterface
 {
     abstract public function filter($queryBuilder, $metadata, $option);
 
-    protected function typeCastField($metadata, $field, $value, $format = null, $doNotTypecastDatetime = false)
+    protected $typeCaster;
+
+    public function __construct($params)
     {
-        if (! isset($metadata->fieldMappings[$field])) {
-            return $value;
+        if (isset($params[1])) {
+            $this->setTypeCaster($params[1]);
         }
+    }
```

Comment:
I don't like this; it makes understanding the positions of arguments paramount, which gets tricky quickly.

In looking at when this is constructed, it's from calling a plugin manager's `get()` method with the optional second argument, `$options`.

The array of options passed from a plugin manager was originally conceived as an _associative array_. That would be one way to fix this here: have the filter manager pass an associative array with named elements, which would allow this to become:

```php
public function __construct(array $params)
{
        $this->setTypeCaster($params['type_caster'] ?? new TypeCaster());
}
```

Reply:
I don't like it as well, but since it was already done like this and I didn't want to distract from the essence of the changes I mimicked the style :-). Will change though!